core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -309,7 +309,7 @@ sealed abstract class IO[+E, +A] extends Serializable { self =>
     new IO.Ensuring(self, finalizer)
 
   /**
-   * Executes the release action only if there was an error.
+   * Executes the release action only if there was a failure value.
```

Comment:
Here is where all kinds of failures lead to cleanup, so I'd say "if this action failed".